utils/testgen.py

```diff
@@ -359,6 +362,10 @@ def pytest_generate_tests(metafunc):
             if not(prov_obj.type_tclass in provider_types or prov_obj.type_name in provider_types):
                 continue
 
+        if provider_keys is not None:
```

Comment:
Oh? I misunderstood the intent of your other comments though, where a test was scoped for type 'openstack' but we wouldn't want to run against all versions of openstack. This PR can probably be closed then, not the bullet we're looking for.